src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,55 @@
+import classnames from 'classnames';
+import { px } from '../../utils';
+import CloseIcon from '../../assets/close.svg?react';
+import Button from '../Button/Button';
+
+export interface ModalProps extends React.HTMLAttributes<HTMLElement> {
+  /**
+   * Boolean that will determine if the modal is open
+   */
+  open?: boolean;
+  /**
+   * Function to close the modal
+   */
+  onClose?: () => void;
+}
+
+const onClosePlaceholder = () => {
+  console.log('On close clicked');
+};
+
+/**
+ * ## Overview
+ *
+ * A component for displaying a modal.
+ *
+ */
+
+const Modal = ({ children, open = false, onClose = onClosePlaceholder, ...props }: ModalProps) => {
+  if (!open) {
+    return null;
+  }
+
+  const buttonClick = (e: React.MouseEvent) => {
+    e.stopPropagation();
+    onClose();
+  };
+
+  return (
+    <div className={classnames(`${px}-modal`)} onClick={onClose} data-testid="modal-background">
+      <div className={classnames(`${px}-modal__container`)} data-testid="modal-container" {...props}>
```

Comment:
we need to extract the classname prop and add it to the top-level className so that if they pass it, it doesn't override our `{$px}-modal` className

```
<div className={classnames(`${px}-modal`, className)} onClick={onClose} data-testid="modal-background" {...props}>
```

In addition, I think we should always spread the props to the top-level component like above, not a child component